components/tidb_query_normal_expr/src/builtin_time.rs

```diff
@@ -557,6 +557,32 @@ impl ScalarFunc {
         let time = Time::from_days(ctx, days)?;
         Ok(Some(Cow::Owned(time)))
     }
+
+    #[inline]
+    pub fn make_date<'a>(
+        &self,
+        ctx: &mut EvalContext,
+        row: &[Datum],
+    ) -> Result<Option<Cow<'a, Time>>> {
+        let mut year = try_opt!(self.children[0].eval_int(ctx, row)) as u32;
+        let mut days = try_opt!(self.children[1].eval_int(ctx, row)) as u32;
```

Comment:
I think it's better to check whether it's a positive number explicitly.

For example, `2014` is a valid year input, while `-4294965282` is not. However, casting `-4294965282` as u32 will result in `2014`.

Reply:
And mysql does not allow days to be zero.

Reply:
`self.children[0].is_unsigned()` and `self.children[0].is_unsigned()` may be helpful.

Reply:
I've added checking whether it's a positive number explicitly.